taxi/taxi_gym_env.py

```diff
@@ -0,0 +1,140 @@
+        self.action_space = gym.spaces.Discrete(len(self.taxi_env.actions))
+
+        #tracking steps + steps per episode + reward behaviour
+        self.T = 0
+        self.max_steps_per_episode = max_steps_per_episode
+        self.ignore_rewards = ignore_rewards
+
+
+
+    def step(self, action):
+        '''
+        step function
+        converts an input from the self.action_space [Discrete space] into a Taxi gridworld move, takes the action on the Taxi gridworld
+
+        input: action (sampled from self.action_space)
+        returns: observation (after taking action), reward, done, info
+        '''
+
+        #increment step counter and take step in TaxiEnv
+        taxi_rendering, reward, is_terminal = self.taxi_env.step(action)
+        taxi_rendering = self._convert_to_int(taxi_rendering)
+
+        self.T += 1
+
+        if self.ignore_rewards:
+            reward = 0
+            done = False
+
+
+        #tracking new info dict + adding terminal state info
+        info = self._get_current_info()
+
+        #consider terminal state v.s. timeout (reaching max steps)
+        timeout = (self.T%self.max_steps_per_episode)==0 if self.max_steps_per_episode is not None else False
+
+        info['timeout'] = timeout
+        info['is_terminal'] = is_terminal
+
+        return taxi_rendering, reward, is_terminal or timeout, info
+
+    def reset(self):
+        '''
+        reset function
+        resets the state to the starting point
+
+        input: None
+        output: state observation after resetting env
+        '''
+
+        #reset taxi environment: returns rendered RGB array
+        rendered_taxi = self.taxi_env.reset(self.have_goal, self.randomize_positions)
+        rendered_taxi = self._convert_to_int(rendered_taxi)
+
+        #reset step count to 0
+        self.T = 0
+
+        return rendered_taxi
+
+    def render(self, mode='rgb_array'):
+        '''
+        render function
+        generates RGB array of environment
+
+        input: None
+        output: RGB array of environment
+        '''
+
+        #render RGB array for environment
+        if mode=='rgb_array':
+            rendered_taxi = self.taxi_env.render()
+            rendered_taxi = self._convert_to_int(rendered_taxi)
+        else:
+            raise NotImplementedError
+
+        return rendered_taxi
+
+
+    def _get_current_info(self):
+        '''return info dictionary for env'''
+
+        info = {}
+
+        #extract state information for info dict
+        state_info = self.taxi_env.get_state()
+
+        taxi_position = state_info[0:2];  info['taxi_x'] = taxi_position[1]; info['taxi_y'] = taxi_position[0]
+
+        #store position + in-taxi information for each passenger
+        for i in range(0,len(state_info[2:]),3):
+
+            p_y, p_x, p_in_taxi = state_info[i:i+3]
+
+            info['p{}_y'.format(i)] = p_y
+            info['p{}_x'.format(i)] = p_x
+            info['p{}_in_taxi'.format(i)] = p_in_taxi
```

Comment:
You got the variable order correct, but I'd like to change the labels `'y' -> 'row'`, `'x' -> 'col'` to match the original and be slightly more descriptive.

https://github.com/camall3n/visgrid/blob/d274e6505fb734a84afdde0e04da23b064824927/taxi/taxi.py#L219-L227

Reply:
Also, for style purposes, let's have one statement per line, no semicolons.